core/src/main/java/org/frankframework/jdbc/dbms/TransactionalDbmsSupportAwareDataSourceProxy.java

```diff
@@ -0,0 +1,173 @@
+		if (metadata == null) {
+			log.debug("populating metadata from getMetaData");
+			try (Connection connection = getConnection()) {
+				populateMetadata(connection);
+			}
+		}
+		return metadata;
+	}
+
+	/**
+	 * Should only be called once, either on the first {@link #getConnection()} or when explicitly requested {@link #getMetaData()}.
+	 */
+	private void populateMetadata(Connection connection) throws SQLException {
+		Map<String, String> databaseMetadata = new HashMap<>();
+		DatabaseMetaData md = connection.getMetaData();
+		databaseMetadata.put("catalog", connection.getCatalog());
+
+		databaseMetadata.put("user", md.getUserName());
+		databaseMetadata.put("url", md.getURL());
+		databaseMetadata.put("product", md.getDatabaseProductName());
+		databaseMetadata.put("product-version", md.getDatabaseProductVersion());
+		databaseMetadata.put("driver", md.getDriverName());
+		databaseMetadata.put("driver-version", md.getDriverVersion());
+
+		metadata = databaseMetadata;
+	}
+
+	public String getDestinationName() throws SQLException {
+		if (destinationName == null) {
+			StringBuilder builder = new StringBuilder();
+			builder.append(getMetaData().get("url"));
+
+			String catalog = getMetaData().get("catalog");
+			if (catalog != null) builder.append("/").append(catalog);
+
+			destinationName = builder.toString();
+		}
+		return destinationName;
+	}
+
+	@Override
+	public Connection getConnection() throws SQLException {
+		Connection conn = super.getConnection();
+		if (metadata == null) {
+			log.debug("populating metadata from getConnection");
+			populateMetadata(conn);
+		}
+		return conn;
+	}
+
+	@Override
+	public String toString() {
+		if (metadata != null && log.isInfoEnabled()) {
+			return getInfo();
+		}
+		return obtainTargetDataSource().toString();
+	}
+
+	public String getInfo() {
+		StringBuilder info = new StringBuilder();
+		if (metadata != null) {
+			info.append("user [").append(metadata.get("user")).append(CLOSE);
+			info.append("url [").append(metadata.get("url")).append(CLOSE);
+			info.append("product [").append(metadata.get("product")).append(CLOSE);
+			info.append("product version [").append(metadata.get("product-version")).append(CLOSE);
+			info.append("driver [").append(metadata.get("driver")).append(CLOSE);
+			info.append("driver version [").append(metadata.get("driver-version")).append(CLOSE);
+		}
+
+		if (getTargetDataSource() instanceof OpenManagedDataSource) {
+			OpenManagedDataSource targetDataSource = (OpenManagedDataSource) getTargetDataSource();
+			GenericObjectPool pool = targetDataSource.getPool();
+			getGenericObjectPoolInfo(pool, info);
+		} else if (getTargetDataSource() instanceof org.apache.commons.dbcp2.PoolingDataSource) {
+			OpenPoolingDataSource dataSource = (OpenPoolingDataSource) getTargetDataSource();
+			GenericObjectPool pool = dataSource.getPool();
+			getGenericObjectPoolInfo(pool, info);
+		} else if (getTargetDataSource() instanceof BasicDataSource) { // Tomcat instance
```

Comment:
When is something a Tomcat instance? That really shouldn't happen.
I strongly propose implementing a default pooling setup in the (default) JndiDataSourceFactory. Else where going to have to create implementations like this for every single pooling mechanism. Also, the tomcat pooling mechanism is far inferior to dbcp2.

Reply:
It can be a Tomcat instance when we use plain tomcat without BTM or Narayana, I believe.
Adding pooling ourselves and managing that is a good idea I think.

Reply:
@nielsm5 I have a draft PR for some connection pooling improvements for DBCP2 pool.

I think to avoid too much overlap between PRs, it might be better that I pick up this in my PR (#6119).

We should then remove all this Tomcat pool handling and the dependency from this PR.

Reply:
As we agreed, closing this comment thread.